openfl/federated/task/runner_xgb.py

```diff
@@ -0,0 +1,382 @@
+            **kwargs: Additional parameters.
+
+        Returns:
+            global_output_dict (dict):  Tensors to send back to the aggregator.
+            local_output_dict (dict):   Tensors to maintain in the local
+                TensorDB.
+        """
+        self.rebuild_model(input_tensor_dict)
+        loader = self.data_loader.get_train_dmatrix()
+        metric = self.train_(loader)
+        # Output metric tensors (scalar)
+        origin = col_name
+        tags = ("trained",)
+        output_metric_dict = {
+            TensorKey(metric.name, origin, round_num, True, ("metric",)): metric.value
+        }
+
+        # output model tensors (Doesn't include TensorKey)
+        output_model_dict = self.get_tensor_dict()
+        global_model_dict, local_model_dict = split_tensor_dict_for_holdouts(
+            self.logger, output_model_dict, **self.tensor_dict_split_fn_kwargs
+        )
+
+        # Create global tensorkeys
+        global_tensorkey_model_dict = {
+            TensorKey(tensor_name, origin, round_num, False, tags): nparray
+            for tensor_name, nparray in global_model_dict.items()
+        }
+        # Create tensorkeys that should stay local
+        local_tensorkey_model_dict = {
+            TensorKey(tensor_name, origin, round_num, False, tags): nparray
+            for tensor_name, nparray in local_model_dict.items()
+        }
+        # The train/validate aggregated function of the next round will look
+        # for the updated model parameters.
+        # This ensures they will be resolved locally
+        next_local_tensorkey_model_dict = {
+            TensorKey(tensor_name, origin, round_num + 1, False, ("model",)): nparray
+            for tensor_name, nparray in local_model_dict.items()
+        }
+
+        global_tensor_dict = {
+            **output_metric_dict,
+            **global_tensorkey_model_dict,
+        }
+        local_tensor_dict = {
+            **local_tensorkey_model_dict,
+            **next_local_tensorkey_model_dict,
+        }
+
+        # Update the required tensors if they need to be pulled from the
+        # aggregator
+        # TODO this logic can break if different collaborators have different
+        # roles between rounds.
+        # For example, if a collaborator only performs validation in the first
+        # round but training in the second, it has no way of knowing the
+        # optimizer state tensor names to request from the aggregator because
+        # these are only created after training occurs. A work around could
+        # involve doing a single epoch of training on random data to get the
+        # optimizer names, and then throwing away the model.
+        if self.opt_treatment == "CONTINUE_GLOBAL":
+            self.initialize_tensorkeys_for_functions()
+
+        # This will signal that the optimizer values are now present,
+        # and can be loaded when the model is rebuilt
+        self.training_round_completed = True
```

Comment:
Given that XGBoost doesn't have discrete optimizer objects, could we skip this code, and the verbose TODO?

Reply:
Yes, you're right. I've removed this bulky portion of code